packages/material-ui/src/Button/Button.test.js

```diff
@@ -281,7 +289,17 @@ describe('<Button />', () => {
   });
 
   it('should render a button with endIcon', () => {
-    const { getByRole } = render(<Button endIcon={<span>icon</span>}>Hello World</Button>);
+    let getByRole;
+
+    expect(() => {
+      const { getByRole: getByRoleLocal } = render(
+        <Button startIcon={<span>icon</span>}>Hello World</Button>,
+      );
+      getByRole = getByRoleLocal;
+    }).toErrorDev(
```

Comment:
@eps1lon am I using the `toErrorDev` correctly here? If I have this expected error in both tests as it's done currently, the tests are failing with https://app.circleci.com/pipelines/github/mui-org/material-ui/30240/workflows/2f2f8de6-1ef0-48e1-9962-9baa4d91821e/jobs/206347

If I remove the `toErrorDev` in the second test, then it is failing with 
```
  19) <Button />
       should render a button with endIcon:
     Expected test not to call console.error()

If the warning is expected, test for it explicitly by using the toErrorDev() matcher.
Test location:
  D:\workspace\material-ui\packages\material-ui\src\Button\Button.test.js

console.error message:
  The pseudo class ":first-child" is potentially unsafe when doing server-side rendering. Try changing it to ":first-of-type".

Stack:
...
```

Should I clear the errors in between somewhere or?

Reply:
Is emotion deduplicating errors at the module level or something?

I wouldn't worry about it since we don't want to ship that warning anyway. SSR should not have any warnings or errors by default.

Reply:
Fixed the warning with https://github.com/mui-org/material-ui/pull/24107/commits/745e163a761e4a34e1f5ac5aa22956e0d22b748b I've replaced the `'& > *:first-child'` selctor with `'& > *:nth-of-type(1)'` to avoid this warning. The result should be identical.

Reply:
> The result should be identical.

More or less 😁 https://codepen.io/oliviertassinari/pen/oNYBeEd